### PR TITLE
WINC-830: Packer refactor

### PR DESCRIPTION
**Pull request by aravindhp**
- [docs] Refactor Packer build steps
- [docs] Migrate Packer build to HCL format 
- [docs] Update golden image creation

### CODE REVIEW

File: docs/vsphere_ci/answer-files/autounattend.xml [hunk 1]
```diff
@@ -113,24 +115,14 @@
...
-                </SynchronousCommand>
             </FirstLogonCommands>
             <OOBE>
                 <HideEULAPage>true</HideEULAPage>
                 <HideLocalAccountScreen>true</HideLocalAccountScreen>
                 <HideOEMRegistrationScreen>true</HideOEMRegistrationScreen>
                 <HideOnlineAccountScreens>true</HideOnlineAccountScreens>
                 <HideWirelessSetupInOOBE>true</HideWirelessSetupInOOBE>
-                <NetworkLocation>Home</NetworkLocation>
+                <NetworkLocation>Work</NetworkLocation>
```
Comment (sebsoto) [inline docs/vsphere_ci/answer-files/autounattend.xml:125]: Does this change have any effect in practice?

File: docs/vsphere_ci/windows-server-2022.pkr.hcl [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    elevated_user     = "${var.vm-elevated-user}"
+    script            = "scripts/install-updates.ps1"
+  }
+
+  # Restart again to apply the updates
+  provisioner "windows-restart" {
+    restart_timeout = "1h"
+  }
+
+  # Pause to allow Windows to caalese and executy a dummy command
```
Comment (sebsoto) [inline docs/vsphere_ci/windows-server-2022.pkr.hcl:136]: Spelling typos

**Review by sebsoto — APPROVED** [review]
Changes mainly LGTM

Will defer to someone who has more recent experience with the packer process

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

I don't see any issues with the approach. I'm interested how you chose the cut-off of what steps should be executed from autounattend and what should be done using provisioners. Is this mixed approach standard?

File: docs/vsphere_ci/answer-files/autounattend.xml (hunk 1 above)
Comment (aravindhp) [reply docs/vsphere_ci/answer-files/autounattend.xml:125]: It is a well known albeit deprecated value.

File: docs/vsphere_ci/windows-server-2022.pkr.hcl (hunk 2 above)
Comment (aravindhp) [reply docs/vsphere_ci/windows-server-2022.pkr.hcl]: Fixed.

**Comment by aravindhp** [comment]
> /lgtm
> 
> I don't see any issues with the approach. I'm interested how you chose the cut-off of what steps should be executed from autounattend and what should be done using provisioners. Is this mixed approach standard?

The cut off was the steps needed to allow Packer to communicate with the template VM. And yes, I have seen this pattern in many repos. Please `/lgtm` again.

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

**Pull request merged**